core/api/src/services/mongoose/quiz.ts

```diff
@@ -0,0 +1,38 @@
+import { Quiz } from "./schema"
+
+import { QuizAlreadyPresentError, UnknownRepositoryError } from "@/domain/errors"
+
+interface ExtendedError extends Error {
+  code?: number
+}
+
+export const QuizRepository = (accountId: AccountId) => {
+  const add = async (quizId: QuizQuestionId) => {
+    try {
+      await Quiz.create({ accountId, quizId })
+
+      return true
+    } catch (err) {
+      if (err instanceof Error) {
+        const error = err as ExtendedError
+        if (error?.code === 11000) return new QuizAlreadyPresentError()
+      }
+
+      return new UnknownRepositoryError("quiz issue")
+    }
+  }
+
+  const fetchAll = async () => {
+    try {
+      const result = await Quiz.find({ accountId })
+      return result
+    } catch (err) {
+      return new UnknownRepositoryError("quiz issue")
+    }
+  }
+
+  return {
```

Comment:
please add tracing wrapper (at least for add)

Reply:
isn't the tracing already provided by mongodb tracing module? we are not doing this for the other module in this mongoose folder